Update configuration-reference.md (#131)
withastro/docs: src/pages/en/reference/configuration-reference.md

The `buildOptions` option configures how a site is built, including its base URL (`buildOptions.site`), whether it includes a sitemap (`buildOptions.sitemap`), whether markdown draft pages should be included in the build (`buildOptions.drafts`), and whether its pages should be files (`path.html`) or directories (`path/index.html`) (`buildOptions.pageUrlFormat`).

**Defaults**:

> - `buildOptions.site`: Used to generate sitemaps and canonical URLs.

Reviewer: I like this. Would this voicing update be acceptable?
```suggestion
- `buildOptions.site`: Public [origin](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin) used to generate sitemaps and canonical URLs.
```

  - Your public domain, e.g.: `https://my-site.dev/`.

> - `buildOptions.sitemap`: Generate an automatically-generated sitemap for your build.

Reviewer: Voicing suggestion.
```suggestion
- `buildOptions.sitemap`: Whether to automatically generate a sitemap.
```

  - Either `true` or `false`.
  - Default: `true`.

> - `buildOptions.pageUrlFormat`: Control the output file URL format of each page.

Reviewer: ```suggestion
- `buildOptions.pageUrlFormat`: Determines how files built from pages are written.
```

  - Either `file` (ex: "/foo.html") or `directory` (ex: "/foo/index.html").
  - Default: `'directory'`.

> - `buildOptions.drafts`: Control if markdown draft pages should be included in the build.

Reviewer: ```suggestion
- `buildOptions.drafts`: Determines whether markdown draft pages are included in the build.
```

  - Either `true` or `false`.
  - Default: `false`.

> Read more about [markdown draft pages][markdown-draft-pages].

Reviewer: ❤️
